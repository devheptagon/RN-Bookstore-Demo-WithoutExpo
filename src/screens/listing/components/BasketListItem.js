import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet } from 'react-native'
import ProductInfo from './ProductInfo'
import FlexView from '../../components/FlexView'
import NumericInput from '../../components/NumericInput'
import { addItem, removeItem } from '../../../services/repos/basketRepo'
import Theme from '../../../constants/Theme'
import { FlexDirections } from '../../../constants/Enums'

export class BasketListItem extends React.PureComponent {
    onIncrease = () => {
        const { dataItem } = this.props
        addItem(dataItem)        
    }

    onDecrease = () => {
        const { dataItem } = this.props
        removeItem(dataItem)
    }

    render () {
        const { dataItem } = this.props        
        return (        
            <FlexView flexDirection={FlexDirections.Row} style={styles.Wrapper}>
                <FlexView flex={1} centerAll style={styles.ButtonContainer}>
                    <NumericInput 
                        value={dataItem.count}                    
                        onIncrease={this.onIncrease}                    
                        onDecrease={this.onDecrease}
                    />
                </FlexView>
                <FlexView flex={6}>
                    <ProductInfo dataItem={dataItem} />
                </FlexView>                
            </FlexView>        
        )
    }
}

const styles = StyleSheet.create({
    Wrapper: {        
        borderTopWidth: Theme.General.Border.Width,
        borderBottomWidth: Theme.General.Border.Width,
        borderTopColor: Theme.ProductList.Item.BorderColor,
        borderBottomColor: Theme.ProductList.Item.BorderColor,
        backgroundColor: Theme.ProductList.Item.BackgroundColor,
    },
    ButtonContainer: {
        borderWidth: Theme.General.Border.Width,                
        borderColor: Theme.Basket.NumericColumn.BorderColor,
        backgroundColor: Theme.Basket.NumericColumn.BackgroundColor
    }
})

BasketListItem.propTypes = {      
    dataItem: PropTypes.object  
}

export default BasketListItem 