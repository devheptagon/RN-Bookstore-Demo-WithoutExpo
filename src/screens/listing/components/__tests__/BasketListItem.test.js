import React from "react";
import { BasketListItem } from "../BasketListItem";
import renderer from "react-test-renderer";

test("BasketListItem renders correctly", () => {
  const tree = renderer.create(<BasketListItem dataItem={{}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
