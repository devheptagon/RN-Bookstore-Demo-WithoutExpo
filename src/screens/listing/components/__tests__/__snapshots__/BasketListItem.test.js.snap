// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketListItem renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomColor": "#f5f5f5",
      "borderBottomWidth": 1,
      "borderTopColor": "#f5f5f5",
      "borderTopWidth": 1,
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderColor": "#eee",
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "height": 100,
          "paddingBottom": 4,
          "paddingTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "width": 32,
            }
          }
        >
          <Text />
        </View>
        <Text
          style={
            Object {
              "color": "green",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          0
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "width": 32,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 6,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "padding": 5,
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "borderColor": "#eee",
                  "borderWidth": 2,
                },
                Object {
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 3,
            "flexDirection": "column",
            "padding": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
